src/sentry_session.c

```diff
@@ -47,7 +47,19 @@ status_from_string(const char *status)
 sentry_session_t *
 sentry__session_new(void)
 {
+    const sentry_options_t *opts = sentry_get_options();
+    if (!opts) {
+        return NULL;
+    }
+    const char *release = sentry_options_get_release(opts);
+    const char *environment = sentry_options_get_environment(opts);
+    if (!release) {
+        return NULL;
+    }
+
     sentry_session_t *rv = SENTRY_MAKE(sentry_session_t);
+    rv->release = release;
```

Comment:
Just pointing out that this is safe because we don't allow to change the release after initialization.